public/src/app.js

```diff
@@ -106,6 +104,42 @@ app.cacheBuster = null;
 		});
 	};
 
+	app.updateHeader = function (data, callback) {
+		/**
+		 * data:
+		 *   header (obj)
+		 *   config (obj)
+		 *   next (string)
+		 */
+		require(['benchpress'], function (Benchpress) {
+			app.user = data.header.user;
+			data.header.config = data.config;
+			config = data.config;
+			Benchpress.setGlobal('config', config);
+
+			// Manually reconnect socket.io
+			socket.close();
+			socket.open();
+
+			// Re-render top bar menu
+			var toRender = [
+				['menu', $('#header-menu .container')],
+				['chats-menu', $('#chats-menu')],
+				['slideout-menu', $('.slideout-menu')],
+			];
+			Promise.all(toRender.map(function (set) {
+				return new Promise(function (resolve) {
```

Comment:
Both translator.translate and benchpress.render return promises. It would probably be better chain those together than use parseAndTranslate like this

Reply:
Can parseAndTranslate return a Promise instead?

Reply:
I guess if you want to change it to do that, go right ahead. Or make a new helper function `renderAndTranslate` that does that.